### DISCUSSION TITLE
mbac in obj

### DISCUSSION THREAD

**Discussion by Vova2124** [minexew/MascotCapsule_Archaeology#13] [Q&A]
Please tell me how to convert mbac in obj

**Comment by minexew** [comment]
mbac2obj.py can be used for the conversion, but it is not built as a user-friendly tool; some proficiency with Python and using the command line is expected. I know nothing about your environment, so I can only guess.

Anyway, on the command line, you would invoke the script like this:
```
python3 mbac2obj.py my_extracted_file.mbac output.obj
```

This will parse the extracted model and generate _output.obj_. (you can use any file names you want, of course)

Note that some games also obfuscate their files to make them more difficult to extract. You would get an "Not a MBAC file" error in that case, but we know how to solve that as well.

**Comment by Vova2124** [comment]
"Exception: Not a MBAC file"

**Reply by minexew** [reply]
Does your model come from a Fishlabs game? In that case, if you first run
```
python3 fishlabs_obfuscation.py my_extracted_file.mbac
```
then you should be able to do the conversion. Note that the de-obfuscation overwrites the original file.

If this doesn't help, you can upload your MBAC somewhere and I will take a look.

**Comment by Vova2124** [comment]
Thank you, everything worked out!

**Reply by minexew** [reply]
My pleasure 👌

**Comment by Vova2124** [comment]
Do you know how to open a file PNG?

**Comment by Vova2124** [comment]
PNG files from this game are not opened by any program

**Reply by woesss** [reply]
Most likely they are encrypted in the same way as `.mbac` - try decrypting with the same `fishlabs_obfuscation.py`

**Comment by Vova2124** [comment]
Thanks!
